lib/dialects/postgres/query-generator.js

```diff
@@ -121,38 +121,108 @@ const QueryGenerator = {
       `WHERE c.table_name = ${this.escape(tableName)} AND c.table_schema = ${this.escape(schema)} `;
   },
 
-  // A recursive parser for nested where conditions
-  parseConditionObject(_conditions, path) {
-    path = path || [];
-    return Utils._.reduce(_conditions, (r, v, k) => { // result, key, value
-      if (Utils._.isObject(v)) {
-        r = r.concat(this.parseConditionObject(v, path.concat(k))); // Recursively parse objects
-      } else {
-        r.push({ path: path.concat(k), value: v });
+  /**
+   * Check whether the statmement is json function or simple path
+   *
+   * @param   {String}  stmt  The statement to validate
+   * @returns {Boolean}       true if the given statement is json function
+   * @throws  {Error}         throw if the statement looks like json function but has invalid token
+   */
+  checkValidJsonStatement(stmt) {
+    if (!_.isString(stmt)) {
+      return false;
+    }
+
+    // https://www.postgresql.org/docs/current/static/functions-json.html
+    const jsonFunctionRegex = /^\s*((?:[a-z]+_){0,2}jsonb?(?:_[a-z]+){0,2})\([^)]*\)/i;
```

Comment:
@janmeier you sure V8 will not optimize this?
@rokoroku I would uppercase it if it's a constant (not if it's a property)